app/src/main/java/space/taran/arkretouch/presentation/edit/EditViewModel.kt

```diff
@@ -137,6 +139,40 @@ class EditViewModel(
         return uri!!
     }
 
+    fun persistUsedColors(context: Context) {
+        viewModelScope.launch(Dispatchers.IO) {
+            try {
+                val writer = context.openFileOutput(
+                    "colors",
+                    Context.MODE_PRIVATE
+                ).bufferedWriter()
+                editManager.oldColors.forEach { color ->
+                    val line = color.value.toString()
+                    writer.appendLine(line)
+                }
+                writer.close()
+            } catch (e: IOException) {
+                e.printStackTrace()
+            }
+        }
+    }
```

Comment:
Extract it into `data/Preferences`, add to di component and pass into `EditViewModel` constructor
Just add `private val prefernces: Preferences` to `EditViewModelFactory` constructor

Reply:
Alright